staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller_test.go

```diff
@@ -0,0 +1,339 @@
+				Pools: map[string]Pool{
+					testPoolName1: {
+						Generation: 1,
+						Devices: []resourceapi.Device{
+							{
+								Name: testDeviceName,
+							},
+						},
+					},
+				},
+			},
+			expectedResourceSlices: nil,
+		},
+	}
+
+	for name, test := range testCases {
+		t.Run(name, func(t *testing.T) {
+			_, ctx := ktesting.NewTestContext(t)
+			// Create kubeClient
+			kubeClient := fake.NewSimpleClientset(test.initialObjects...)
+
+			// Create controller
+			ctrl := setupController(ctx, testNode, test.testNodeUID, testDriveName,
+				kubeClient, test.inputDriverResources)
+
+			err := ctrl.syncPool(ctx, test.inputPoolName)
+			assert.Equal(t, test.expectedError, err, "unexpected error")
+
+			// Check ResourceSlices if meets expectations
+			resourceSlices, err := kubeClient.ResourceV1alpha3().ResourceSlices().List(ctx, metav1.ListOptions{})
+			require.NoError(t, err, "list resource slices")
+
+			if test.expectedResourceSlices != nil {
+				assert.Equal(t, test.expectedResourceSlices, resourceSlices.Items, "unexpected ResourceSlices")
+			}
+		})
+	}
+}
+
+// setupController sets up the controller with the given parameters.
+func setupController(ctx context.Context, testNode string, testNodeUID types.UID,
+	testDriveName string, kubeClient *fake.Clientset, inputDriverResources *DriverResources) *Controller {
+	// Set up the test environment
+	_, cancel := context.WithCancelCause(ctx)
+
+	// Create the Owner object
+	owner := Owner{
+		APIVersion: "v1",
+		Kind:       "Node",
+		Name:       testNode,
+		UID:        testNodeUID,
+	}
+
+	// Create the Controller instance
+	ctrl := &Controller{
+		cancel:     cancel,
+		kubeClient: kubeClient,
+		driver:     testDriveName,
+		owner:      owner,
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig(
+			workqueue.DefaultTypedControllerRateLimiter[string](),
+			workqueue.TypedRateLimitingQueueConfig[string]{Name: "node_resource_slices"},
+		),
+		mutex: sync.RWMutex{},
+	}
+
+	// Set up informerFactory and sliceStore
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(kubeClient, 0,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			// Set filtering criteria
+			selector := fields.Set{resourceapi.ResourceSliceSelectorDriver: ctrl.driver}
+			if ctrl.owner.APIVersion == "v1" && ctrl.owner.Kind == "Node" {
+				selector[resourceapi.ResourceSliceSelectorNodeName] = ctrl.owner.Name
+			}
+			options.FieldSelector = selector.String()
+		}))
+
+	resourceSliceInformer := informerFactory.Resource().V1alpha3().ResourceSlices().Informer()
+	err := resourceSliceInformer.AddIndexers(cache.Indexers{
+		poolNameIndex: func(obj interface{}) ([]string, error) {
+			slice, ok := obj.(*resourceapi.ResourceSlice)
+			if !ok {
+				return []string{}, nil
+			}
+			return []string{slice.Spec.Pool.Name}, nil
+		},
+	})
+	if err != nil {
+		return nil
+	}
+
+	// Start informer factory
+	informerFactory.Start(ctx.Done())
+	informerFactory.WaitForCacheSync(ctx.Done())
```

Comment:
```suggestion
	informerFactory.WaitForCacheSync(ctx.Done())
	t.Cleanup(informerFactory.Shutdown)
```

`t` must be passed in for this. Otherwise each `setupController` call leaks the factory goroutines.

Reply:
After splitting, directly using the informer inside the controller, it seems that this operation is no longer necessary